backtesting/backtesting.py

```diff
@@ -1339,7 +1339,7 @@ def _compute_stats(self, broker: _Broker, strategy: Strategy) -> pd.Series:
 
         equity = pd.Series(broker._equity).bfill().fillna(broker._cash).values
         dd = 1 - equity / np.maximum.accumulate(equity)
-        dd_dur, dd_peaks = self._compute_drawdown_duration_peaks(pd.Series(dd, index=data.index))
```

Comment:
@kernc I was having some issues with Max DD not showing correctly in the stats and in the plots You can see for yourself in the legend MAX DD is 0 days. https://www.realitysharesadvisors.com/fund_dash/fix_this.html  

![image](https://user-images.githubusercontent.com/32274861/97064254-79858c00-1559-11eb-8eb3-39e94ce7ca9b.png)


So I saw this change in the diff for recent pr #156 I added this back and now it works.  `index=data.index` instead of `index=index` 

That fixed the calculation, but the red line is misplaced in plots still for some reason so I'm still digging into that.

Here is an example. 
https://www.realitysharesadvisors.com/fund_dash/fix_this_eth.html

Reply:
> I added this back and now it works. `index=data.index` instead of `index=index`

I don't understand it, `index` is set to the same value just a couple lines prior:
https://github.com/kernc/backtesting.py/blob/191f4753b2a405b20e30add2267afc69a5b06858/backtesting/backtesting.py#L1343-L1347
and `_compute_drawdown_duration_peaks()` doesn't seem to touch it in any destructive way. :open_mouth: :confused:

> but the red line is misplaced in plots still for some reason so I'm still digging into that.

Can you verify this works on master/released version? If not, I'd suspect `Backtest.plot(resample=True)`:
https://github.com/kernc/backtesting.py/blob/191f4753b2a405b20e30add2267afc69a5b06858/backtesting/backtesting.py#L1479-L1489

Reply:
Okay, so you were absolutely right, the stats were not the problem.  I'm sorry for not catching that before posting.  The drawdown calculations were fine in the _compute_stats.  The legend in the plot disagrees with the stats and I didn't pick up on that. 

**Testing**: I created a new clean environment and pip installed the master/released version of backtesting.py.  When it is run on daily plots or other plots with fewer than 10,000 bars, the equity plot correctly shows the max drawdown in the correct location as well as the red bar across the proper peak/valley/peak.  

However, when it is run on plots that need to be resampled it comes up with some erroneous results.  The 0-day drawdown for example, or the drawdown period over a different section than the max drawdown.  "red dot"

Also of note, though I don't think it matters for the drawdown calculations. I set resample=False in bt.plot(resample=False) and it appears to have ignored the suggestion. 

I'm not sure but is the following if statement ignoring the False boolean? I believe Bools are a subclass of int so checking for str skips down to the "else" statement and reassigns freq to one of the resample periods.  Again, if I'm just causing more trouble than it's worth I'm happy to just blindly point out issues and not even bother trying to identify the culprits. 

`def _maybe_resample_data(resample_rule, df, indicators, equity_data, trades):
    if isinstance(resample_rule, str):
        freq = resample_rule
    else:
        if...`

Reply:
> or the drawdown period over a different section than the max drawdown. "red dot"

The max drawdown _period length_ is fully distinct from the max drawdown _value_. But your plots indicate clearly the computations are amiss.

> I'm not sure but is the following if statement ignoring the False boolean? I believe Bools are a subclass of int so checking for str skips down to the "else" statement and reassigns freq to one of the resample periods. Again, if I'm just causing more trouble than it's worth I'm happy to just blindly point out issues and not even bother trying to identify the culprits.

Oops. You're perfectly right. Fixed in https://github.com/kernc/backtesting.py/commit/05329deaa507e64dbd7858f9d253eecb20ab9f16.

I think the one thing everyone likes even better than issues is fixes for them. :stuck_out_tongue_closed_eyes: 

-----

I gather these are unrelated to this PR then? Would you mind opening a new issue for that invalid resampled drawdown?